aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -111,11 +116,15 @@ public OutputFile newOutputFile(String path) {
 
   @Override
   public void deleteFile(String path) {
-    S3URI location = new S3URI(path);
```

Comment:
to minimize diff, we can do the reverse logic instead to avoid a lot of indentation change, e.g.

```
if (!deleteTags.isEmpty()) {
  doSoftDelete(path);
  return;
}

// original logic
...
```

Reply:
Sure, I have made the changes.